llm4pddl/llm_interface.py

```diff
@@ -108,13 +112,27 @@ def _sample_completions(
             temperature: float,
             seed: int,
             num_completions: int = 1) -> List[LLMResponse]:  # pragma: no cover
-        response = openai.Completion.create(
-            model=self._model_name,  # type: ignore
-            prompt=prompt,
-            temperature=temperature,
-            max_tokens=self._max_tokens,
-            logprobs=1,
-            n=num_completions)
+        # Always max out the max tokens to get the longest possible responses.
+        num_prompt_tokens = len(self._tokenizer(prompt)["input_ids"])
+        # If the prompt is too long, warn and give up.
+        max_response_tokens = FLAGS.llm_max_total_tokens - num_prompt_tokens
+        if max_response_tokens <= 0:
+            logging.warning("Prompt length exceeded token budget, skipping!")
+            return []
+        # Repeatedly query and sleep to deal with rate limit.
+        while True:
+            try:
+                response = openai.Completion.create(
+                    model=self._model_name,  # type: ignore
+                    prompt=prompt,
+                    temperature=temperature,
+                    max_tokens=max_response_tokens,
+                    logprobs=1,
+                    n=num_completions)
+                # Successfully got a response.
+                break
+            except openai.error.RateLimitError:
+                time.sleep(5)  # sleep for 5 seconds and then try again
```

Comment:
shouldnt this be 3 at most for codex since its 20/min? and 1 for GPT-3 since it is 60/min?

Reply:
yeah I ended up removing this for now because it wasn't helping, I was still getting throttled when everything was parallelized. fingers crossed for a rate limit increase, otherwise i'll just do it single threaded (in which case each query takes far longer than 3 seconds, so we're not hitting the limit)